### PR TITLE
Stop testing against ancient versions of numpy

### PR DESCRIPTION
**Pull request by drdavella**
This is an attempt to resolve some travis errors that are cropping up on recent tests. It's worth asking whether we need to be testing on Py3.4 at all, especially since recent versions of `gwcs` explicitly do not support anything earlier than 3.5.

### CODE REVIEW

**Comment by brechmos-stsci** [comment]
The question about py34 is likely relevant for errors cropping up in https://github.com/astropy/specutils/pull/270 too.

**Comment by drdavella** [comment]
This should be merged soon since there are quite a few other PRs that would need to be rebased in order for tests to pass.

**Comment by eteq** [comment]
The 3.4 error may be *real* errors - there's an issue that the doctests haven't been running on astropy >= 3.0 lately, but the 3.4 build is running that. (#307). Am working on a fix for that, but in the meantime maybe just add a astropy==2.0 test (but not bother with the ancient numpu/Python)

**Comment by drdavella** [comment]
@eteq just to be clear: do you want me to remove Py3.4 tests altogether, but add astropy 2.x starting with Py3.5?

**Comment by eteq** [comment]
@drdavell yep exactly. I think it's safe to drop 3.4 aupport guarantees given the fairly small user base to begin with

File: .travis.yml [hunk 1]
```diff
@@ -43,9 +43,9 @@ env:
 
     matrix:
         # Make sure that egg_info works without dependencies
-        - PYTHON_VERSION=3.4 SETUP_CMD='egg_info'
         - PYTHON_VERSION=3.5 SETUP_CMD='egg_info'
         - PYTHON_VERSION=3.6 SETUP_CMD='egg_info'
+        - PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
```
Comment (bsipocz) [inline .travis.yml]: you can drop all but one of the egg_info builds, too.

File: .travis.yml [hunk 2]
```diff
@@ -72,11 +72,13 @@ matrix:
         # time.
 
         - os: linux
-          env: PYTHON_VERSION=3.4 NUMPY_VERSION=1.9
+          env: PYTHON_VERSION=3.5
```
Comment (bsipocz) [inline .travis.yml]: this one can be dropped as well I think, it should be enough to test python3.5 once in the matrix

File: .travis.yml (hunk 1 above)
Comment (drdavella) [reply .travis.yml]: Which one should I keep? 3.7?

Comment (bsipocz) [reply .travis.yml]: It shouldn't matter, the one you like the most :)

**Comment by bsipocz** [comment]
@drdavella - One more thing. Since those failures looked very much real, and it's also otherwise a good idea, could you please add one more job to the matrix, to test with `ASTROPY_VERSION=dev`?

**Pull request merged**